lib/url.js

```diff
@@ -353,6 +353,13 @@ function urlFormat(obj) {
   // this way, you can call url_format() on strings
   // to clean up potentially wonky urls.
   if (typeof obj === 'string') obj = urlParse(obj);
+
+  if (typeof obj !== 'object')
```

Comment:
@bnoordhuis Does that also match for situations like eg. how the code is now which is `else if (typeof obj !== 'object' || obj === null)`?